modules/swagger-codegen/src/main/java/io/swagger/codegen/DefaultCodegen.java

```diff
@@ -1975,20 +1975,20 @@ protected void setNonArrayMapProperty(CodegenProperty property, String type) {
      * @param responses Swagger Operation's responses
      * @return default method response or <tt>null</tt> if not found
      */
-    protected Response findMethodResponse(Map<String, Response> responses) {
+    protected Entry<String, Response> findMethodResponse(Map<String, Response> responses) {
```

Comment:
If I understand your code correctly, this seems to be a change to the CORE of swagger codegen since it change the way we handle return status. At the beginning, I though it was only Play Framework generator that was missing feature to return the good code but since you needed to add stuff in the DefaultCodegen, I guess it was missing from all codegen. 

That is making this change request more important and I think that core members should review it too.

@wing328  Can you check this and also tag some core members to review this.

Thanks!